lib/asciidoctor/opal_ext/file.rb

```diff
@@ -19,12 +19,19 @@ class File
   attr_reader :lineno
   attr_reader :path
 
-  def initialize(path, mode = 'r')
+  def initialize(path, flags = 'r')
     @path = path
     @contents = nil
     @eof = false
     @lineno = 0
-    @fd = `require('fs').openSync(path, mode)` if JAVASCRIPT_PLATFORM == 'node'
+    # binary flag is unsupported
+    binary_flag_regexp = /b/
```

Comment:
If you read the image file with the encoding set to 'utf8', you get corrupt data. (I'm not sure if that's what the code is doing. I'm just mentioning that a general comment).